kasa/cli/discover.py

```diff
@@ -30,15 +35,33 @@
 
 @click.group(invoke_without_command=True)
 @click.pass_context
-async def discover(ctx):
+async def discover(ctx: click.Context):
     """Discover devices in the network."""
     if ctx.invoked_subcommand is None:
         return await ctx.invoke(detail)
 
 
+@discover.result_callback()
+@click.pass_context
+async def _close_protocols(ctx: click.Context, discovered: DeviceDict):
+    """Close all the device protocols if discover was invoked directly by the user."""
```

Comment:
Good catch on this! We should probably also do something similar when `error()`ing out (as we currently get those warnings from aiohttp), but that's something for a separate PR.